sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -275,10 +285,19 @@ export class EventHubClient {
   }
 
   /**
-   * Creates an EventHubProducer that can be used to send events to the Event Hub for which this client
-   * was created.
+   * Creates an Event Hub producer responsible for transmitting `EventData` to the Event Hub.
+   * Depending on the `options` specified, the producer may be created to allow event data to
```

Comment:
> Depending on whether a partitionId is specified in the `options`, ....

This makes the association of the presence/absence of partitionId and the outcome more clear